test/IDE/complete_in_closures.swift

```diff
@@ -371,3 +371,28 @@ func testInsideTernaryClosureReturn(test: Bool) -> [String] {
         // SINGLE_TERNARY_EXPR_CLOSURE_CONTEXT: End completions
     }
 }
+
+func testSignature() {
+    func accept<T>(_: () -> T) {}
+
+    accept { #^PARAM_BARE_1?check=EMPTY^# in }
+    accept { #^PARAM_BARE_2?check=EMPTY^#, arg2 in }
+    accept { arg1, #^PARAM_BARE_3?check=EMPTY^# in }
+
+    accept { (#^PARAM_PAREN_1?check=EMPTY^#) in }
+    accept { (#^PARAM_PAREN_2?check=EMPTY^#, arg2) in }
+    accept { (arg1, #^PARAM_PAREN_3?check=EMPTY^#) in }
+
+    accept { (arg1: #^PARAMTYPE_1?check=WITH_GLOBAL_DECLS^#) in }
+    accept { (arg1: Int, arg2: #^PARAMTYPE_2?check=WITH_GLOBAL_DECLS^#) in }
+
+    accept { [#^CAPTURE_1?check=WITH_GLOBAL_DECLS^#] in }
+    accept { [weak #^CAPTURE_2?check=WITH_GLOBAL_DECLS^#] in }
+    accept { [#^CAPTURE_3?check=EMPTY^# = capture] in }
```

Comment:
This should offer `weak`/`unowned` too. But because it needs some plumbing work, I'd like to do it in followups.